source/kernel/modules.go

```diff
@@ -42,3 +45,30 @@ func getLoadedModules() ([]string, error) {
 	}
 	return loadedMods, nil
 }
+
+func getBuiltinModules() ([]string, error) {
+	kVersion, err := getVersion()
+	if err != nil {
+		return []string{}, err
+	}
+
+	kBuiltinModPath := hostpath.LibDir.Path("modules/" + kVersion + "/modules.builtin")
+	out, err := os.ReadFile(kBuiltinModPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read file %s: %s", kBuiltinModPath, err.Error())
+	}
+
+	lines := strings.Split(string(out), "\n")
+	builtinMods := make([]string, len(lines))
```

Comment:
You need to set the lenght of the array to zero, otherwise it will have no use here (it will have `len(lines)` number of empty strings on which you will append the new values, making the final slice 2 x `len(lines)` long)

```suggestion
	builtinMods := make([]string, 0, len(lines))
```

Reply:
@marquiz should we change it also for the `loadedMods := make([]string, len(lines))` (It's in the function above)?

Reply:
No worries, I'll include it here.